### PR TITLE
feat: Synchronize CRs Reasons

### PR DESCRIPTION
**Pull request by TeodorSAP**
## Description

Changes proposed in this pull request (what was done and why):

- New names and structure for the reasons used
   - Rename `Deployment` to `Gateway`
   - Rename `Daemon` to `Agent`
   - Delete CR prefixes (e.g. Log, Metric, Trace, Pipeline, etc.) and prefixing logic
   - Merge `TLSCertificateInvalid` reasons
- Un-hardcode `LogComponentsHealthy`, `TraceComponentsHealthy`, and `MetricComponentsHealthy` condition types
- Adapt documentation according to changes, and add missing reasons

Changes refer to particular issues, PRs or documents:

- #1023 

## Traceability
- [x] The PR is linked to a GitHub issue.
- [x] New features have a milestone set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [x] The feature is e2e-tested

### CODE REVIEW

**Comment by NHingerl** [comment]
Should this be on the Huskies board?

File: docs/user/resources/01-telemetry.md [hunk 1]
```diff
@@ -109,9 +109,8 @@ The state of the log components is determined by the status condition of type `L
...
-| False            | LogPipelineReferencedSecretMissing | One or more referenced Secrets are missing                                                                                                                                                                                                                |
-| False            | FluentBitDaemonSetNotReady         | Fluent Bit DaemonSet is not ready                                                                                                                                                                                                                         |
+| True             | ComponentsRunning               | All log components are running                                                                                                                                                                                                                            |
```
Comment (a-thaler) [inline docs/user/resources/01-telemetry.md:112]: Not really related but aren't there many possible reasons missing like TLSCertificateExpired?

File: docs/user/resources/02-logpipeline.md [hunk 2]
```diff
@@ -199,18 +199,16 @@ The status of the LogPipeline is determined by the condition types `AgentHealthy
...
+| ConfigurationGenerated | False            | ReferencedSecretMissing     | One or more referenced Secrets are missing                                                                                                                                                                                          |
+| ConfigurationGenerated | False            | UnsupportedLokiOutput       | grafana-loki output is not supported anymore. For integration with a custom Loki installation, use the `custom` output and follow [Intergrate with Loki](https://kyma-project.io/#/telemetry-manager/user/integration/loki/README). |
+| ConfigurationGenerated | False            | TLSCertificateInvalid       | TLS certificate invalid                                                                                                                                                                                                             |
```
Comment (a-thaler) [inline docs/user/resources/02-logpipeline.md:209]: In which reason you will cover "TLSCertificateKeyPairInvalid"?

Comment (TeodorSAP) [reply docs/user/resources/02-logpipeline.md:209]: These two, together with "TLSCertificateInvalid" will be all covered in "TLSCertificateInvalid"

**Review by NHingerl — APPROVED** [review]

File: internal/conditions/conditions.go [hunk 3]
```diff
@@ -18,34 +21,34 @@ const (
...
-	ReasonTLSCertificateAboutToExpire  = "TLSCertificateAboutToExpire"
-	ReasonTLSCertificateKeyPairInvalid = "TLSCertificateKeyPairInvalid"
-
-	ReasonMetricAgentNotRequired  = "AgentNotRequired"
-	ReasonMetricComponentsRunning = "MetricComponentsRunning"
-
+	// Universal pipeline reasons
+	ReasonNoPipelineDeployed      = "NoPipelineDeployed"
+	ReasonComponentsRunning       = "ComponentsRunning"
+	ReasonReferencedSecretMissing = "ReferencedSecretMissing"
```
Comment (skhalash) [inline internal/conditions/conditions.go:27]: ReferencedSecretMissing is a common pipeline reason, the rest are Telemetry only reasons (not universal)

File: docs/user/resources/02-logpipeline.md [hunk 4]
```diff
@@ -199,23 +199,21 @@ The status of the LogPipeline is determined by the condition types `AgentHealthy
...
+| ConfigurationGenerated | False            | TLSCertificateExpired       | TLS certificate expired on YYYY-MM-DD                                                                                                                                                                                               |
+| ConfigurationGenerated | True             | TLSCertificateAboutToExpire | TLS certificate is about to expire, configured certificate is valid until YYYY-MM-DD                                                                                                                                                |
```
Comment (skhalash) [inline docs/user/resources/02-logpipeline.md:211]: Those tables are randomly sorted, which makes them harder to read. I suggest we sort them:
1. By the status (True, then False)
2. If the status is the same, then alphabetically

**Review by skhalash — CHANGES_REQUESTED** [review]

File: docs/user/resources/02-logpipeline.md (hunk 4 above)
Comment (skhalash) [reply docs/user/resources/02-logpipeline.md:211]: I also mean the other md files

**Review by skhalash — APPROVED** [review]

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**